core/src/main/java/org/elasticsearch/common/xcontent/support/AbstractXContentParser.java

```
@@ -289,7 +289,7 @@ public BytesRef utf8BytesOrNull() throws IOException {
             token = parser.nextToken();
         }
         ArrayList<Object> list = new ArrayList<>();
-        for (; token != XContentParser.Token.END_ARRAY; token = parser.nextToken()) {
+        for (; token != null && token != XContentParser.Token.END_ARRAY && token != XContentParser.Token.END_OBJECT; token = parser.nextToken()) {
```

could we throw an exception instead then?